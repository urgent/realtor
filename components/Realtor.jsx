import React from "react";
import { Box, Image, Heading, Text } from "@chakra-ui/react";

export function Realtor({ image, name, address }) {
  return (
    <Box
      data-cy="realtor-row"
      borderWidth="1px"
      borderRadius="lg"
      overflow="hidden"
    >
      <Image data-cy="realtor-image" src={image} />
      <Box>
        <Heading as="h3" data-cy="realtor-name">
          {name}
        </Heading>
        <Text data-cy="realtor-address">{address}</Text>
      </Box>
    </Box>
  );
}

export function RealtorTable({ realtors }) {
  return (
    <Box data-cy="realtor-table">
      {realtors.map((realtor, index) => (
        <Realtor key={index} {...realtor} />
      ))}
    </Box>
  );
}
