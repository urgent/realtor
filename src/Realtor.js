import React from 'react'
import { Card, Flex, Box, Image, Heading, Text } from 'rebass/'

function Realtor(props) {
    return <Card>
        <Flex >
            <Box p="3">
                <Image src="/bio/Georgeanna-Newmones.jpg" />
            </Box>
            <Box p="3">
                <Heading fontSize="2">Georgeanna Newmones</Heading>
                <Text>Executive Realtors</Text>
                <Text>3817 Ventnor Ave</Text>
                <Text>Atlantic City, NJ 8401</Text>
                <Text>Office - 609-645-0303</Text>
            </Box>
        </Flex>
    </Card>
}

export default Realtor