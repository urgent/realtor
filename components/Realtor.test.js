import React from "react";
import renderer from "react-test-renderer";
import { Realtor, RealtorTable } from "./Realtor";

const data = [
  {
    name: "Georgeanna Newmones",
    address:
      "Executive Realtors\r\n3817 Ventnor Ave\r\nAtlantic City, NJ 8401\r\nOffice - 609-645-0303",
    image: "/bio/Georgeanna-Newmones.jpg",
  },
];

it("renders correctly", () => {
  const realtor = renderer.create(<Realtor {...data[0]} />).toJSON();
  expect(realtor).toMatchSnapshot();
  const table = renderer.create(<RealtorTable realtors={data} />).toJSON();
  expect(table).toMatchSnapshot();
});
