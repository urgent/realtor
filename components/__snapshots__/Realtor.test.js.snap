// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1oqd3nr"
  data-cy="realtor-row"
>
  <img
    className="chakra-image__placeholder css-0"
    data-cy="realtor-image"
  />
  <div
    className="css-0"
  >
    <h3
      className="chakra-heading css-0"
      data-cy="realtor-name"
    >
      Georgeanna Newmones
    </h3>
    <p
      className="chakra-text css-0"
      data-cy="realtor-address"
    >
      Executive Realtors
3817 Ventnor Ave
Atlantic City, NJ 8401
Office - 609-645-0303
    </p>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="css-0"
  data-cy="realtor-table"
>
  <div
    className="css-1oqd3nr"
    data-cy="realtor-row"
  >
    <img
      className="chakra-image__placeholder css-0"
      data-cy="realtor-image"
    />
    <div
      className="css-0"
    >
      <h3
        className="chakra-heading css-0"
        data-cy="realtor-name"
      >
        Georgeanna Newmones
      </h3>
      <p
        className="chakra-text css-0"
        data-cy="realtor-address"
      >
        Executive Realtors
3817 Ventnor Ave
Atlantic City, NJ 8401
Office - 609-645-0303
      </p>
    </div>
  </div>
</div>
`;
